neptune/new/attributes/atoms/artifact.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2021, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pathlib
+from typing import List
+
+from neptune.new.internal.artifacts.types import ArtifactFileData, ArtifactDriversMap
+from neptune.new.attributes.atoms.atom import Atom
+
+
+class Artifact(Atom):
+    def fetch_hash(self) -> str:
+        val = self._backend.get_artifact_attribute(self._run_uuid, self._path)
+        return val.hash
+
+    def fetch_files_list(self) -> List[ArtifactFileData]:
+        artifact_hash = self.fetch_hash()
+        return self._backend.list_artifact_files(self._run.get_project_identifier(), artifact_hash)
+
+    def download(self, destination: str = None):
+        for file_definition in self.fetch_files_list():
+            driver = ArtifactDriversMap.match_type(file_definition.type)
+            driver.download_file(
+                pathlib.Path(destination) / file_definition.file_path, file_definition
```

Comment:
According to your suggestions in `s3.py:82` it should be only "destination". https://github.com/neptune-ai/neptune-client/pull/639/files#diff-4d67da793bba3029a8bf7bd1eaa0b317b1c32c651d4ba5df857e572996f2f62dR82 https://github.com/neptune-ai/neptune-client/pull/644#discussion_r685833069